### ISSUE TITLE
fix: Not able update subtitle when switch on second video

### ISSUE THREAD

**Issue by androrahul** [Joran-Dob/flutter_subtitle_wrapper#84]
**Description**

**Steps To Reproduce**

**Expected Behavior**

**Screenshots**

**Additional Context**

**Comment by androrahul** [comment]
I have a video list that works fine on the first video, but when I switch to another video, I still see the old subtitles.


  return FlickVideoPlayer(
      flickManager: flickManager!,
      flickVideoWithControls: SubtitleWrapper(
        videoPlayerController:
        flickManager!.flickVideoManager!.videoPlayerController!,
        subtitleController: subtitleController!,
        subtitleStyle: SubtitleStyle(
            textColor: Colors.white,
            hasBorder: true,
            fontSize: 10.sp,
            position: const SubtitlePosition(bottom: 45)),
        videoChild: const FlickVideoWithControls(
          controls: FlickPortraitControls(),
        ),
      ),
      flickVideoWithControlsFullscreen: SubtitleWrapper(
        videoPlayerController:
        flickManager!.flickVideoManager!.videoPlayerController!,
        subtitleController: subtitleController!,
        subtitleStyle: SubtitleStyle(
            textColor: Colors.white,
            hasBorder: true,
            fontSize: 14.sp,
            position: const SubtitlePosition(bottom: 65)),
        videoChild: const FlickVideoWithControls(
          controls: FlickLandscapeControls(),
          videoFit: BoxFit.fitHeight,
        ),
      ),
    );

reseting second video:

void resetVideoPlayer(List<LessonDetailsList> lessonDetailsList, int index) {
    if (lessonDetailsList[index].videoLibrary != null) {

      if(videoPlayerController != null){
        subtitleController!.detach();
        videoPlayerController!.pause();
        videoPlayerController!.dispose();
        videoPlayerController = null;
        subtitleController = null;
        // flickVideoPlayer = null;
        // flickManager!.dispose();
      }

      videoPlayerController = VideoPlayerController.networkUrl(Uri.parse(
          "https://${Constant.main}/video?fileName=${lessonDetailsList[index].videoLibrary!.videoFileName!}"))..initialize().then((_) {
        setState(() {

          videoPlayerController!.addListener(checkVideo);

          if (lessonDetailsList[index].userEnrollmentLesson != null) {
            final newPosition = getVideoDuration(
                lessonDetailsList[index]
                    .userEnrollmentLesson!
                    .currentPlayTime!);

            videoPlayerController!.seekTo(newPosition);
          }
        });
      });

      subTitleList.clear();

      lessonId = lessonDetailsList[index].id!;

      if (lessonDetailsList[index].videoLibrary!.videoSubtitleEn !=
          null) {

        subtitleController =  SubtitleController(
            subtitleUrl: SwConstants.enSubtitle,
            subtitleDecoder: SubtitleDecoder.utf8,
            subtitleType: SubtitleType.srt);

        // subtitleController!.updateSubtitleUrl(url: "https://${Constant.main}/lessonSubtitles?fileName=${lessonDetailsList[index].videoLibrary!.videoSubtitleEn!}");

        if(kDebugMode) print("https://${Constant.main}/lessonSubtitles?fileName=${lessonDetailsList[index].videoLibrary!.videoSubtitleEn!}");

        // updateSubtitleUrl(
        //     "https://${Constant.main}/lessonSubtitles?fileName=${lessonDetailsList[index].videoLibrary!.videoSubtitleEn!}");

        subTitleList.add(
            "https://${Constant.main}/lessonSubtitles?fileName=${lessonDetailsList[index].videoLibrary!.videoSubtitleEn!}");
      } else {
        // updateSubtitleUrl("");
      }

      if (lessonDetailsList[index].videoLibrary!.videoSubtitleAr !=
          null) {
        subTitleList.add(
            "https://${Constant.main}/lessonSubtitles?fileName=${lessonDetailsList[index].videoLibrary!.videoSubtitleAr!}");
      }

      if(kDebugMode) print("subTitleList: ${subTitleList.length}");

      if(flickManager != null){
        flickManager!.handleChangeVideo(videoPlayerController!);
      } else{
        flickManager = FlickManager(
          videoPlayerController: videoPlayerController!,
        );
      }


    } else {
      Utility.showToast("No Video found.");
    }
  }

**Comment by Joran-Dob** [comment]
I'm having a bit of trouble understanding the code you provided. Would you mind forking this project and giving me a reproducible example in the fork? That way, I can take a closer look.

**Comment by Joran-Dob** [comment]
Not sure if this is still an problem for you, if so feel free to reopen!
